SimCalorimetry/EcalSimAlgos/interface/EcalElectronicsSim_Ph2.h

```diff
@@ -0,0 +1,57 @@
+#ifndef SimCalorimetry_EcalSimAlgos_EcalElectronicsSim_Ph2_h
+#define SimCalorimetry_EcalSimAlgos_EcalElectronicsSim_Ph2_h
+
+#include "CalibFormats/CaloObjects/interface/CaloTSamples_Ph2.h"
+#include "SimCalorimetry/CaloSimAlgos/interface/CaloVNoiseSignalGenerator.h"
+#include "CondFormats/EcalObjects/interface/EcalConstants.h"
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+
+class EcalCoder_Ph2;
+//class EcalDataFrame_Ph2;
+class EcalSimParameterMap_Ph2;
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+/* \class EcalElectronicsSim_Ph2
+ * \brief Converts CaloDataFrame in CaloTimeSample and vice versa.
+ * 
+ */
+
+class EcalElectronicsSim_Ph2 {
+public:
+  typedef CaloTSamples_Ph2<float, ecalPh2::sampleSize> EcalSamples;
```

Comment:
similarly to others, the original `EcalElectronicsSim` class can be used with templates for the samples and coder classes